api_docgen/Makefile

```diff
@@ -12,20 +12,33 @@
 #*   special exception on linking described in the file LICENSE.          *
 #*                                                                        *
 #**************************************************************************
+# Used by included Makefiles
 ROOTDIR = ..
--include $(ROOTDIR)/Makefile.build_config
-
-ifeq ($(DOCUMENTATION_TOOL),odoc)
-  include odoc/Makefile
-else
-  include ocamldoc/Makefile
-endif
+-include ../Makefile.build_config
 
 odoc-%:
-	$(MAKE) -C odoc $* ROOTDIR=../..
+	$(MAKE) -C odoc $*
 
 ocamldoc-%:
-	$(MAKE) -C ocamldoc $* ROOTDIR=../..
+	$(MAKE) -C ocamldoc $*
+
+ifeq ($(DOCUMENTATION_TOOL),odoc)
+man: odoc-man
+latex: odoc-latex
+html: odoc-html
+	@echo "documentation is in ./api_docgen/odoc/build/html/"
+install: odoc-install
+else
+man: ocamldoc-man
+latex: ocamldoc-latex
+html: ocamldoc-html
+	@echo "documentation is in ./api_docgen/ocamldoc/build/html/"
+install: ocamldoc-install
+texi: ocamldoc-texi
+pdf: ocamldoc-pdf
+endif
```

Comment:
Since the main Makefile is not including `Makefile.common` anymore, you probably to define it explicitly here
```
all: html pdf man
```

Reply:
The targets are a bit different depending on the backend, I've added the `all` target but it's not at the top. Should it be?

Reply:
The `all` target is more a test target, it should not be at the top.